File: src/game/tools/rpc/output/wrap/ShopItemListDataWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class ShopItemListDataWrap : IWrapper
	{	
		public ShopItemDataWrap[] ShopItemList; // 商店item数据
		public long NextTime; // 下次跟新时间
		
		public void Decode(Packet pck)
		{	
			ShopItemList = new ShopItemDataWrap[pck.GetShort()];
			for (int i = 0; i < ShopItemList.Length; i++)
			{
				ShopItemList[i] = new ShopItemDataWrap();
				ShopItemList[i].Decode(pck);
				
			}
			
			NextTime = pck.GetLong(); 
		}
		
        public void Encode(Packet pck)
        {	
        	if (ShopItemList == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)ShopItemList.Length);
	        	for(int i = 0; i < ShopItemList.Length; i++)
	        	{
	        		ShopItemList[i].Encode(pck);
					
	        	}
	        }
        	
        	pck.PutLong(NextTime); 
        }
	}
}

File: src/game/tools/rpc/output/wrap/ShopItemListDataWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type ShopItemListDataWrap struct {
	
	ShopItemList []*ShopItemDataWrap // 商店item数据
	NextTime int64 // 下次跟新时间
	rpc.Wrapper
}

func (w *ShopItemListDataWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.ShopItemList = make([]*ShopItemDataWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.ShopItemList); i++ {
		w.ShopItemList[i] = new(ShopItemDataWrap)
		w.ShopItemList[i].Decode(pck)
		
	}
	
	w.NextTime = pck.PopInt64(); 
	return w
}

func (w *ShopItemListDataWrap)Encode(pck *rpc.Packet) {
	
	if w.ShopItemList == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.ShopItemList)))
    	for i := 0; i < len(w.ShopItemList); i++ {
    		w.ShopItemList[i].Encode(pck);
			
    	}
    }
	
	pck.PutInt64(w.NextTime) 
}

